File: 1201-1300/1299-replace-elements.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        # loop from the last element, set max_value as -1 initially
        # we will set max_value to be the current element if current element
        # has higher value
        max_value = -1
        for i in range(len(arr) - 1, -1, -1):
            # store the current value in a temp variable
            temp = arr[i]
            
            # update current element first as we need to replace
            # every element with the greatest element to its right,
            # even the current element is greater
            arr[i] = max_value
            
            # update max value if temp is greater
            if max_value < temp:
              max_value = temp
        return arr
            

File: 1201-1300/1299-replace-elements.java
class Solution {
  public int[] replaceElements(int[] arr) {
      
      int highest = arr[arr.length-1];
      
      // set the last element to -1
      arr[arr.length-1] = -1;
      
      // start from the second last element
      for(int i=arr.length-2;i>=0;i--) {
          int temp = highest;
          
          if(arr[i] > highest) {
              highest = arr[i];
          }
          
          arr[i] = temp;
      }
      
      return arr;
  }
}